### ISSUE TITLE
DATETIMEVALUEFMT formula with "[Fn,-3], [D] [Mn,-3] [Y0001] [H01]:[m01]:[s01] [Z0000]" format

### ISSUE THREAD

**Issue by ghprud** [IBM/JSONata4Java#227]
When using the following formula

DATETIMEVALUEFMT formula with "[Fn,-3], [D] [Mn,-3] [Y0001] [H01]:[m01]:[s01] [Z0000]" format
Test string: "Tue, 4 Oct 2022 12:00:01 -0400"

I see the following error
JSONata syntax error: line 2:0 at [@0,1:0='<EOF>',<-1>,2:0]: mismatched input '<EOF>' expecting {'(', 'true', 'false', STRING, 'null', '[', '
{', '$', '$$', '**', NUMBER, FUNCTIONID, '*', '/', '-', '%', VAR_ID, ID}
JSONata mapping error: For input string: ""

**Comment by wnm3** [comment]
I was hoping to recreate the problem but wasn't sure which functions you were calling. 

Where you calling something like the examples in this part of the documentation? https://docs.jsonata.org/date-time

The formats allowed in jsonata.org are based on these descriptions from XPath: https://www.w3.org/TR/xpath-functions-31/#func-format-dateTime

Perhaps if you could show the expression you were passing into the Tester or TesterUI utility?

**Comment by wnm3** [comment]
It could be the input date format is not ISO 8601 format.

This works on try.jsonata.org
$toMillis('2022-10-04T12:00:01.000-0400')~>$fromMillis("[Fn,-3], [D] [Mn,-3] [Y0001] [H01]:[m01]:[s01] [Z0000]")

However this doesn't pass through JSONata4Java. I believe there is a problem in fromMillis that won't work with the timezone on the string.

These work:
$toMillis('2022-10-04T12:00:01')
$toMillis('2022-10-04T12:00:01.000')
But this fails:
$toMillis('2022-10-04T12:00:01.000-0400')

**Comment by ygenin** [comment]
Yes, the error happens with the following which works on JSONata:

$fromMillis($toMillis(**"Tue, 4 Oct 2022 12:00:01 -0400"**, **"[Fn,-3], [D] [Mn,-3] [Y0001] [H01]:[m01]:[s01] [Z0000]"** ))
